src/UI/class-settings-page.php

```diff
@@ -598,6 +612,30 @@ public function print_multiple_checkboxes( array $args ): void {
 		$this->print_description_text( $args );
 	}
 
+	/**
+	 * Print out a "single-image browse control" which includes a text
+	 * input to store image path and a button to browse for images.
+	 *
+	 * @param array $args The arguments for the control.
+	 * @return void
+	 */
+	public function print_media_single_image( array $args ) {
+		$key         = esc_attr( $args['option_key'] );
+		$input_value = $this->parsely->get_options()[ $key ];
+		$input_name  = Parsely::OPTIONS_KEY . "[$key]";
+		$button_text = __( 'Browse', 'wp-parsely' );
+		?>
+
+		<fieldset class="media-single-image" id="media-single-image-<?php echo esc_attr( $key ); ?>">
+			<legend class="screen-reader-text"><span><?php echo esc_html( $args['title'] ); ?></span></legend>
+			<input class="large-text file-path" type="text" name="<?php echo esc_attr( $input_name ); ?>" value="<?php echo esc_attr( $input_value ); ?>" />
+			<button data-option="<?php echo esc_attr( $key ); ?>" class="browse button" style="margin-top: 0.4rem" type="button"><?php echo esc_html( $button_text ); ?></button>
```

Comment:
Not a fan of inline styles either but I made this exception because this is really a single statement and I didn't think this would warrant a file of its own. Having said that:

- I can move it to wp-parsely.css.
- I can put this in a wp-admin CSS file that would be useful if we're thinking of introducing more styling in the admin in the future.

Another approach (that WP employs but IMO find worse from all the above) would be to put the button in a `<p>` tag. That's what WP does for the "Save Changes" button in settings. But it doesn't require external files or inline styles. 😄

Reply:
I like the `<p>` idea, actually.

Reply:
Done! I've also updated the PR's screenshot since the spacing is a bit less.